python/mxnet/contrib/onnx/mx2onnx/_op_translations.py

```diff
@@ -1655,3 +1678,45 @@ def convert_size(node, **kwargs):
     and return the created node.
     """
     return create_basic_op_node('Size', node, kwargs)
+
+@mx_op.register("_sample_multinomial")
+def convert_multinomial(node, **kwargs):
+    """Map MXNet's multinomial operator attributes to onnx's
+    Multinomial operator and return the created node.
+    """
+    name, input_nodes, attrs = get_inputs(node, kwargs)
+    dtype = onnx.mapping.NP_TYPE_TO_TENSOR_TYPE[np.dtype(attrs.get("dtype", 'int32'))]
+    sample_size = convert_string_to_list(attrs.get("shape", '1'))
+    if len(sample_size) < 2:
+        sample_size = sample_size[-1]
+    else:
+        raise AttributeError("ONNX currently supports integer sample_size only")
+    node = onnx.helper.make_node(
+        "Multinomial",
+        input_nodes,
+        [name],
+        dtype=dtype,
+        sample_size=sample_size,
+        name=name,
+    )
+    return [node]
+
+
+@mx_op.register("take")
+def convert_gather(node, **kwargs):
+    """Map MXNet's size_array operator attributes to onnx's Size operator
+    and return the created node.
+    """
+    name, input_nodes, attrs = get_inputs(node, kwargs)
+
+    axis = int(attrs.get('axis', '0'))
```

Comment:
How is the behavior of mode attribute of Take op in MXNet change with Gather operator in ONNX? Do we support all modes like clip/raise/wrap?

Reply:
ONNX's documentation is not very clear about this, but their test comapre with np.take(), so it looks like an error should be raised when the indices are out of bounds. MXNet doesn't support 'raise' mode yet 

> MXNetError: [10:36:05] src/operator/tensor/./indexing_op.h:688: Raise is not supported for the time being...

Reply:
Export of Gather will be revisited later when MXNet backend supports 'raise' mode in Take operator